### PR TITLE
"Empty State" for Invited Members

### LINKED ISSUE #17451: Invited Members: "Empty State" for Invited Members
📖 **As an Admin, I want my experience with invitations to be as pleasing and straightforward as possible, so that I feel motivated to use them.**

✅ Tasks to complete:
- [x] Add an empty state for Invited members

### PR DESCRIPTION
**Pull request by juliannatetreault**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR adds an "empty state" to the invited members table, which shows when there are no invitations pending.

## Related Tickets & Documents
- Closes #17451 

## QA Instructions, Screenshots, Recordings
To QA this PR, navigate to `/admin/invitations` and ensure that when there are no invited members, that you are shown the "empty state":
#### After:
Screen Shot 2022-05-04 at 2 54 21 PM

### UI accessibility concerns?
Hopefully not!

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

John Travolta in Pulp Fiction looking around

### CODE REVIEW

File: app/views/admin/invitations/index.html.erb [hunk 1]
```diff
@@ -80,8 +81,13 @@
             <%= render "admin/users/index/invitation_actions_dropdown", user: user, context: "large" %>
           </td>
         </tr>
+      </tbody>
+      <% end %>
+      <% else %>
+      <td>
+        <%= render "admin/invitations/empty_state" %>
```
Comment (juliannatetreault) [inline app/views/admin/invitations/index.html.erb:88]: So, I'm running into something interesting here... I am unable to place the "empty state" where I need it (centered within the table body, beneath the table headings, per the design). I am continually running into a situation where the "empty state" is placed above the table headings, centered and looking as expected, _or_ it is within the table body, but is not centered and does is not looking as expected--I've included screenshots below for reference:

Placed above the table headings, centered and looking as expected:
Screen Shot 2022-05-04 at 1 44 22 PM

Within the table body, but not centered or looking as expected:
Screen Shot 2022-05-04 at 1 44 02 PM

Any thoughts on how to get around this? It's clearly the table that is causing issues, as the "empty state" looks as expected when it renders outside of the table columns/rows, but the design shows the "empty state" within the table, beneath the table headers. 🤔

File: app/views/admin/invitations/index.html.erb [hunk 2]
```diff
@@ -66,20 +66,24 @@
...
+            </td>
+            <td><%= user.email %></td>
+            <td><%= user.created_at.strftime("%d %b, %Y") %></td>
+            <td>
+              <%= render "admin/users/index/invitation_actions_dropdown", user: user, context: "large" %>
+            </td>
+          </tr>
+        <% end %>
+        <% else %>
+          <%= render "admin/invitations/empty_state" %>
```
Comment (juliannatetreault) [inline app/views/admin/invitations/index.html.erb:86]: So, I'm running into something interesting here... I am unable to place the "empty state" where I need it (centered within the table body, beneath the table headings, per the design). I am continually running into a situation where the "empty state" is placed above the table headings, centered and looking as expected, _or_ it is within the table body, but is not centered and does is not looking as expected--I've included screenshots below for reference:

**Placed above the table headings, centered and looking as expected:**
Screen Shot 2022-05-04 at 1 44 22 PM

**Within the table body, but not centered or looking as expected:**
Screen Shot 2022-05-04 at 1 44 02 PM

Any thoughts on how to get around this? I've tried different stylings, moving the "empty state" around in the view, removing and adding table rows, etc. to no avail. It's clearly the table that is causing issues, as the "empty state" looks as expected when it renders outside of the table columns/rows, but the design shows the "empty state" within the table, beneath the table headers. 🤔

Comment (jeremyf) [reply app/views/admin/invitations/index.html.erb:86]: Quick question, is the empty state partial re-used?  If it is where is it used?  If not let's consider inlining it for this exercise.

My assumption is that we'd have something like:

```html
<tr>
  <td colspan="4">
    <div class="align-center flex flex-col my-auto py-7">
       <p class="fs-xl fw-normal mb-2">No members invited yet.</p>
       <p class="color-secondary">Invite members and grow your community.</p>
    </div>
    <div class="flex grow-1 justify-content-center">
       <%= link_to "Invite member", new_admin_invitation_path, class: "c-cta c-cta--branded ml-3" %>
    </div>
  </td>
</tr>
```

Looking at MDN docs for tables, the allowed elements are `caption`, `colgroup`, `thead`, `tbody`, `tr` and `tfoot` (see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table)

So, you'll need a `tr` tag.  Which ([per MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr)) needs a `td` tag.  The `td` tag can then have flow elements (see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/td); the flow content are `div` and `p`-tags (and more).

Comment (aitchiss) [reply app/views/admin/invitations/index.html.erb]: Just catching up with this now, and I think what we want to do is have the empty state message _outside_ of the table completely. Putting the empty state inside the table itself causes conflicting info for users of assistive tech - for the example above we'd announce we had 1 row of data in the table, and the invited member username is "No members invited yet..." etc.

TL;DR, let's move this empty state chunk to below the table itself. As a bonus we could even link the empty state message with the table by giving it an ID and then using `aria-describedby="empty-state-message-id` on the `<table>` element 🙂

Comment (juliannatetreault) [reply app/views/admin/invitations/index.html.erb]: Sounds good; will do!

File: app/views/admin/invitations/index.html.erb [hunk 3]
```diff
@@ -56,7 +56,7 @@
     </div>
 
   </header>
-  <table class="hidden m:table crayons-table crayons-table--minimal" width="100%">
+  <table class="hidden m:table crayons-table crayons-table--minimal" width="100%" aria-describedby="empty-state-message-id">
```
Comment (aitchiss) [inline app/views/admin/invitations/index.html.erb:59]:
```suggestion
  <table class="hidden m:table crayons-table crayons-table--minimal" width="100%" <%= @invitations.empty? ?  ' aria-describedby="empty-state-message-id"'.html_safe : '' %>">
```

I don't think there's any negative impact, but we'll see errors in things like axe dev tools if we reference IDs that don't exist on the page (which will be the case when the empty state message isn't shown)

**Review by aitchiss — APPROVED** [review]
Looks so much better with this empty state!

File: app/views/admin/invitations/index.html.erb (hunk 3 above)
Comment (juliannatetreault) [reply app/views/admin/invitations/index.html.erb:59]: This is a great suggestion; thanks, @aitchiss!

**Review by Ridhwana — APPROVED** [review]
🎉
